File: font/glyph/i.js
import { CAP, MID, SBL, SBR, WIDTH, XH } from '../const'
import { mix } from '../math'
import { corner, endOpen, open, simpleGlyph, spiro } from '../outline'
import { rect, ringAt } from '../shape'

const strokeW = 0.17 * WIDTH
const halfStrokeW = 0.5 * strokeW

const strokeH = XH
const dotR = 0.65 * strokeW

export const i = simpleGlyph(
  spiro(
    open(MID, strokeW),
    corner(MID, strokeH - halfStrokeW),
    endOpen(mix(MID, SBL, 1 / 5), strokeH - halfStrokeW),
  ).stroked(strokeW),
  rect(SBL, halfStrokeW, SBR, halfStrokeW, strokeW),
  ringAt(MID, CAP - dotR, dotR),
)

File: crates/monoxide-font/src/font/glyph/i.rs
use monoxide_script::prelude::*;

use crate::{
    InputContext,
    font::{glyph::l::LShape, math::mix, prelude::*, shape::Ring},
};

pub fn i(cx: &InputContext) -> Glyph {
    let settings = cx.settings();
    let FontParamSettingsView {
        sbl, sbr, stw, xh, ..
    } = settings.view();

    Glyph::builder()
        .outlines(LShape::new(sbl..sbr, 0.0..xh).stroked(stw))
        .outline(dot(settings))
        .build()
}

pub fn i_cap(cx: &InputContext) -> Glyph {
    let FontParamSettingsView {
        sbl, sbr, stw, cap, ..
    } = cx.settings().view();

    Glyph::builder()
        .outlines(
            LShape::new(sbl..sbr, 0.0..cap)
                .with_top_bar_scale((-1.)..1.)
                .stroked(stw),
        )
        .build()
}

pub fn dot(settings: &FontParamSettings) -> Ring {
    let FontParamSettingsView {
        cap, dtr, mid, sbl, ..
    } = settings.view();
    Ring::at((mix(mid, sbl, 0.97), 0.97 * cap), (dtr, dtr))
}
